Initial submission of Final Project

krystleta opened the pull request:
I had accidentally worked on the main branch before I created a new branch for stage-react-api therefore main is almost identical to stage-react-api, I just updated the readme file to be able to make a pull request.

ImaliaNsk commented:
Amazing job, the project looks awesome! Some issues require fixing, but they are mostly pretty straightforward.

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.

* In navigation menu, the active link should be "underilned", please check the design.

- [ ] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.

* home page scrolls on 320 - 418
* saved-newa page scrolls on 320 - 569
* https://snipboard.io/bz5PJ1.jpg Cards should be centered on all resolutions

- [ ] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.

* Footer needs working links, opening in new tabs

- [ ] <!--4.2--> Classes are named according to BEM specifications.

* https://snipboard.io/5JxXUv.jpg Elements can't be used outside of their block. Please fix all similar issues in the project

- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.

* Each page needs one h1
* Text elements can't be `div`, please use headings and paragraphs for text content: `savednews__header-title` and similar issues. Please check and fix. Almost all elements on the page are inside div tags.
* news cards list should be a list `<ul>`, or/and the card needs to be `<article>`
* Links in `header` and `footer` should be in nav, ul or ul inside nav.

- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [ ] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.

* Search button needs active state

- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [ ] <!--1.66--> Raster and vector images have been optimized.

* Please use raster graphics for images, and vector for icons. For example, background image doesn't need to be a vector.

## Number of points: 50

krystleta commented:
Thank you for your feedback!

- In navigation menu, the active link should be "underlined", please check the design. - done
- home page scrolls on 320 - 418 - fixed
- saved-news page scrolls on 320 - 569 - fixed
- https://snipboard.io/bz5PJ1.jpg Cards should be centered on all resolutions - fixed
- Footer needs working links, opening in new tabs - fixed
- Elements can't be used outside of their block. Please fix all similar issues in the project - fixed
- Each page needs one h1
- Text elements can't be div, please use headings and paragraphs for text content: savednews__header-title and similar issues. -fixed 
- news cards list should be a list <ul>, or/and the card needs to be <article> - fixed
- Links in header and footer should be in nav, ul or ul inside nav. - in UL
- Search button needs active state - fixed
- Please use raster graphics for images, and vector for icons. For example, background image doesn't need to be a vector. - fixed

ImaliaNsk commented:
Almost there, only one point short! You did a great job, not much left to fix. Keep up the good work!

# Summary:
### Performance Criteria

- [x] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.
- [ ] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.

* saved-news page scrolls on 481 - 545
* https://snipboard.io/XPamtK.jpg Cards should be centered on all resolutions

- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [ ] <!--4.2--> Classes are named according to BEM specifications.

* https://snipboard.io/5JxXUv.jpg Elements can't be used outside of their block. Please fix all similar issues in the project

- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.

* Each page needs one h1
* Text elements can't be `div`, please use headings and paragraphs for text content: `savednews__header-title` and similar issues. Please check and fix. Almost all elements on the page are inside div tags.
* `<ul>` for news articles is used incorrectly: each card should be a list item. Now you have only one list item containing all news.
* Links in `header` and `footer` should be in nav, ul or ul inside nav.

- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [ ] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [x] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 79

krystleta commented:
- saved-news page scrolls on 481 - 545 - I cannot seem to reproduce this scroll -https://paste.pics/ROFUI
- https://snipboard.io/XPamtK.jpg Cards should be centered on all resolutions - fixed
- <ul> for news articles is used incorrectly: each card should be a list item. Now you have only one list item containing all news. - fixed
- Each page needs one h1 - fixed
- Text elements can't be div, please use headings and paragraphs for text content: savednews__header-title and similar issues. Please check and fix. Almost all elements on the page are inside div tags. - fixed

ImaliaNsk commented:
Congratulations, the project is accepted! I wish you success in your future projects!

# Summary:
### Performance Criteria

- [x] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.
- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [ ] <!--4.2--> Classes are named according to BEM specifications.

* https://snipboard.io/5JxXUv.jpg Elements can't be used outside of their block. Please fix all similar issues in the project

- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.

* Each page needs only one h1. Headings of cards can't be h1.

- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [ ] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [x] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.
- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [x] <!--1.66--> Raster and vector images have been optimized.

## Number of points: 92
